src/App.tsx

```diff
@@ -1,26 +1,198 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { addTodos, deleteTodos, getTodos, USER_ID } from './api/todos';
+import { Header } from './components/header';
+import { TodoList } from './components/todoList';
+import { Footer } from './components/footer';
+import { Error } from './components/Error';
+import { Todo } from './types/Todo';
+import { Filter } from './types/Filter';
 
 export const App: React.FC = () => {
+  const newTodoInputRef = useRef<HTMLInputElement>(null);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [newTodo, setNewTodo] = useState<string>('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [newFilter, setNewFilter] = useState<Filter>(Filter.All);
+  const [isActive] = useState<number>();
+  const todosLeft = todos.filter(todo => !todo.completed).length;
+  const [isLoading, setIsLoading] = useState(false);
+  const [todoClear, setTodoClear] = useState<boolean>(false);
+  const [isInputDisabled, setIsInputDisabled] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [deletingTodoId, setDeletingTodoId] = useState<number | null>(null);
+
+  const loadTodos = async () => {
+    setIsLoading(true);
+    setErrorMessage('');
+
+    try {
+      const todosData = await getTodos();
+      const completedTodos = todosData.filter(todo => todo.completed);
+
+      setTodos(todosData);
+      setTodoClear(completedTodos.length > 0);
+    } catch (error) {
+      setErrorMessage('Unable to load todos');
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    loadTodos().then(() => {
+      newTodoInputRef.current?.focus();
+    });
+  }, []);
+
+  useEffect(() => {
+    setTodoClear(todos.some(todo => todo.completed));
+    newTodoInputRef.current?.focus();
+  }, [todos]);
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setErrorMessage('');
+    }, 3000);
+
+    return () => clearTimeout(timer);
+  }, [errorMessage]);
+
+  function deleteTodo(todoId: number) {
```

Comment:
It is important to stick to one way of creating functions, fix it everywhere
```suggestion
  const deleteTodo = (todoId: number) => {
```